source/slang/parameter-binding.cpp

```diff
@@ -321,29 +366,59 @@ LayoutSemanticInfo ExtractLayoutSemanticInfo(
         break;
 
     default:
-        // TODO: issue an error here!
+        getSink(context)->diagnose(semantic->registerName, Diagnostics::unknownRegisterClass, registerClassName);
         return info;
     }
 
-    // TODO: need to parse and handle `space` binding
-    int space = 0;
+    // For a `register` semantic, the register index is not optional (unlike
+    // how it works for varying input/output semantics).
+    if( registerIndexDigits.size() == 0 )
+    {
+        getSink(context)->diagnose(semantic->registerName, Diagnostics::expectedARegisterIndex, registerClassName);
+    }
 
     UInt index = 0;
-    for (UInt ii = 1; ii < registerName.Length(); ++ii)
+    for(auto c : registerIndexDigits)
     {
-        int c = registerName[ii];
-        if (c >= '0' && c <= '9')
-        {
```

Comment:
Theres this section, and the section further down that do this conversion. Could just have the function in one place (say in StringUtil ;)) And have it test for an error. int StringUtil::parsePositiveInt(), return <0 on error (say overflow, wrong character etc).

Reply:
I think that the `String` type already has some methods like that, but I'm one of those folks that tends to just write stuff from scratch using raw C if I can't remember an alternative that does just what I want without allocating memory, etc.